x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -370,6 +461,24 @@ public boolean verify(SecureString text, char[] hash) {
             return CharArrays.constantTimeEquals(Base64.getUrlEncoder().withoutPadding().encodeToString(md.digest()).toCharArray(), hash);
         }
     },
+    /*
+     * Unsalted SHA-512 , not suited for password storage.
+     */
+    SHA512() {
```

Comment:
I don't think we need this introduced here as part of the ENUM. There is no use for it and it might confuse someone to actually use it for password hashing while they should not. I suggest a private method for the hashing ( we don't need the verification part ) and calling that from the `hash()` and `verify()` of the newly introduced hasher.  We should probably remove the SHA256 from here and I shouldn't have added it in the first place. ( Can do the latter in an unrelated PR ) 

Unless we feel there is need/value in exposing unsalted SHA256 / SHA512 for caching hash algorithms but I don't think we should.

Reply:
Changed